src/installed_dune_file.ml

```diff
@@ -0,0 +1,57 @@
+open Import
+
+let parse_sub_systems sexps =
+  List.filter_map sexps ~f:(fun sexp ->
+    let name, ver, data =
+      Sexp.Of_sexp.(triple string (located string) raw) sexp
+    in
+    match Sub_system_name.get name with
+    | None ->
+      (* We ignore sub-systems that are not internally known. These
+         correspond to plugins that are not in use in the current
+         workspace. *)
+      None
+    | Some name -> Some (name, (Sexp.Ast.loc sexp, ver, data)))
+  |> Sub_system_name.Map.of_list
+  |> (function
+    | Ok x -> x
+    | Error (name, _, (loc, _, _)) ->
+      Loc.fail loc "%S present twice" (Sub_system_name.to_string name))
+  |> Sub_system_name.Map.mapi ~f:(fun name (_, version, data) ->
+    let (module M) = Jbuild.Sub_system_info.get name in
+    let vloc, ver = version in
+    if M.version <> ver then
+      Loc.fail vloc "Unsupported version, only version %s is supported"
```

Comment:
I'd include `ver` here as well. Even though it's accessible via `vloc`, it's still an annoying roundtrip.

Reply:
This code has moved to a new `Syntax.Versioned_parser` module and I included the version number in the message